### PR TITLE
chore: Remove the deprecation of statefulMapConcat operator.

### PR DESCRIPTION
**Pull request by He-Pin**
Motivation:
As reported in refs:https://github.com/apache/incubator-pekko/issues/1141 , the current `statefulMap` will introduce some overhead from the allocation of tuple `(State, Out)`. So let's just keep it before we come up with an allocation free version like JDK 22's gather api.

Result:
Remove the deprecation of `statefulMapConcat`.

Will need backport to 1.0.x branch.

Ping @JD557

### CODE REVIEW

**Review by pjfanning — APPROVED** [review]
lgtm

**Review by pjfanning — CHANGES_REQUESTED** [review]
https://github.com/apache/incubator-pekko/blob/c619f0fe59e52b4ff5407b01d8f0636d40a4df26/docs/src/main/paradox/stream/operators/Source-or-Flow/statefulMapConcat.md?plain=1#L12 needs to be changed too

**Comment by He-Pin** [comment]
@pjfanning PTAL, I will backport this once got merged.

**Comment by laglangyue** [comment]
LGTM

File: docs/src/main/paradox/stream/operators/Source-or-Flow/statefulMapConcat.md
Comment (pjfanning) [inline docs/src/main/paradox/stream/operators/Source-or-Flow/statefulMapConcat.md:9]: I think I might be able to come up with a fuller description later today or tomorrow. Could we hold off on merging?

Comment (He-Pin) [reply docs/src/main/paradox/stream/operators/Source-or-Flow/statefulMapConcat.md:9]: Please change directly on this pr

**Review by pjfanning — APPROVED** [review]
lgtm - I think the warning change is ok

**Pull request merged**
